tests/projects/test_remote_pipeline.py

```diff
@@ -187,8 +196,121 @@ def test_kfp_pipeline_enriched_with_affinity_and_tolerations_enriched_by_preempt
                         raise mlrun.errors.MLRunRuntimeError(
                             "You missed a container to test"
                         )
-        # remove generated workflow file
-        os.remove(workflow_path)
+
+    def test_kfp_pipeline_enriched_with_security_context(
```

Comment:
This test is quite long, please either break it up into separate tests (success flow and the relevant error flows) or use `@pytest.mark.parametrize` to set the `enrichment_mode`, `kfp_pod_user_unix_id`, `expected_security_context`, `expect_error`, etc. for all the different cases and have the test dynamically expect what it needs.